facade-test/createVM_test.go

```diff
@@ -0,0 +1,324 @@
+package facadetest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	prismclientv4facade "github.com/nutanix-cloud-native/prism-go-client/facade/v4"
+	"github.com/nutanix-cloud-native/prism-go-client/internal/testhelpers"
+	"github.com/nutanix-cloud-native/prism-go-client/utils"
+	prismclientv3 "github.com/nutanix-cloud-native/prism-go-client/v3"
+	clustermgmtconfig "github.com/nutanix/ntnx-api-golang-clients/clustermgmt-go-client/v4/models/clustermgmt/v4/config"
+	vmmconfig "github.com/nutanix/ntnx-api-golang-clients/vmm-go-client/v4/models/vmm/v4/ahv/config"
+	vmmcontent "github.com/nutanix/ntnx-api-golang-clients/vmm-go-client/v4/models/vmm/v4/content"
+	"k8s.io/utils/ptr"
+)
+
+var (
+	v4FacadeClient *prismclientv4facade.FacadeV4Client
+	v3Client       *prismclientv3.Client
+)
+
+func initializeClients(t *testing.T) error {
+	var err error
+	credentials := testhelpers.CredentialsFromEnvironment(t)
+	fmt.Printf("Using credentials for host: %s", credentials.Endpoint)
+
+	v3Client, err = prismclientv3.NewV3Client(credentials)
+	if err != nil {
+		t.Errorf("Error creating v3 client: %v", err)
+	}
+
+	v4FacadeClient, err = prismclientv4facade.NewFacadeV4Client(credentials)
+	if err != nil {
+		t.Errorf("Error creating facade v4 client: %v", err)
+	}
+
+	return nil
+}
+
+func TestCreateVm(t *testing.T) {
+	err := initializeClients(t)
+	if err != nil {
+		t.Errorf("failed to intialize prism clients, error: %v", err)
+	}
+
+	response, err := v3Client.V3.ListAllCluster(context.Background(), "")
```

Comment:
There is ListAllClusters method in v4 facade. Please use it.

Reply:
You can use _ListClusterVirtualGPUs_ and _ListClusterPhysicalGPUs_ to check if GPUs are present